packages/node_modules/@webex/plugin-logger/src/logger.js

```diff
@@ -47,7 +47,7 @@ const authTokenKeyPattern = /[Aa]uthorization/;
 function walkAndFilter(object, visited = []) {
   if (visited.includes(object)) {
     // Prevent circular references
-    return '[Circular]';
+    return object;
```

Comment:
Originally I was going to fix this by checking `object` with `isObject(object)` and only recording the visit in such a case. That also required a check of `!isFunction(object)` to fix

> uncaughtException TypeError: Function.prototype.apply was called on [Circular], which is a string and not a function

so the code looked like

    if (isObject(object) && !isFunction(object)) {
      if (visited.includes(object)) {
        ......

But when I gave it more thought I realised it is possible for a function itself to hold a circular reference (though I'm not sure if that will happen).

I also realised I originally returned `[Circular]` as I believed the objected (which is cloned when sent to the logger) would never be used. Turns out that is not the case.

So now instead of setting `[Circular]` the code just immediately returns the object if it has already been visited. That will still filter out sensitive information, while allowing the object to remain usable, and prevent an unending recursive loop.